src/main/java/io/numaproj/numaflow/batchmapper/DatumIterator.java

```diff
@@ -0,0 +1,20 @@
+package io.numaproj.numaflow.batchmapper;
+
+
+/**
+ * An iterator over a collection of {@link Datum} elements.
+ * Passed to {@link BatchMapper#processMessage(DatumIterator)} (DatumIterator)} method.
+ */
+public interface DatumIterator {
+
+    /**
+     * Returns the next element in the iterator
+     * This method blocks until an element becomes available in the blocking queue.
+     * When EOF is received, this method will return null and the iterator will be closed.
+     *
+     * @return the next element in the iterator
+     *
+     * @throws InterruptedException if the thread is interrupted while waiting for the next element
+     */
+    Datum next() throws InterruptedException;
```

Comment:
blocking queue is the implementation detail of the interface hence should not be exposed in the method description.

```suggestion
     * Returns the next element in the iterator
     * This method blocks until an element becomes available.
     * When EOF_DATUM is received, this method will return null and the iterator will be closed.
     *
     * @return the next element in the iterator, null if EOF_DATUM is received or the iterator is already closed
     *
     * @throws InterruptedException if the thread is interrupted while waiting for the next element
     */
    Datum next() throws InterruptedException;
```

Reply:
Please also update it for sinker interface.

Reply:
done